datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -356,8 +356,11 @@ impl BuiltInWindowFunctionExpr for WindowUDFExpr {
     }
 
     fn field(&self) -> Result<Field> {
-        let nullable = true;
-        Ok(Field::new(&self.name, self.data_type.clone(), nullable))
+        Ok(Field::new(
+            &self.name,
+            self.data_type.clone(),
+            self.fun.nullable(),
```

Comment:
This is why `nullable` method was added to `WindowUDFImpl` so that the user-defined window function can customize nullability of column values.